File: imagepicker/src/main/java/com/nguyenhoanglam/imagepicker/ui/imagepicker/ImagePickerView.java
package com.nguyenhoanglam.imagepicker.ui.imagepicker;

import com.nguyenhoanglam.imagepicker.model.Asset;
import com.nguyenhoanglam.imagepicker.model.Folder;
import com.nguyenhoanglam.imagepicker.ui.common.MvpView;

import java.util.List;

/**
 * Created by hoanglam on 8/17/17.
 */

public interface ImagePickerView extends MvpView {

    void showLoading(boolean isLoading);

    void showFetchCompleted(List<Asset> assets, List<Folder> folders);

    void showError(Throwable throwable);

    void showEmpty();

    void showCapturedAsset(List<Asset> assets);

    void finishPickAssets(List<Asset> assets);

}

File: imagepicker/src/main/java/com/nguyenhoanglam/imagepicker/ui/imagepicker/ImagePickerView.kt
package com.nguyenhoanglam.imagepicker.ui.imagepicker

import com.nguyenhoanglam.imagepicker.model.Folder
import com.nguyenhoanglam.imagepicker.model.Image
import com.nguyenhoanglam.imagepicker.ui.common.MvpView

/**
 * Created by hoanglam on 8/17/17.
 */
interface ImagePickerView : MvpView {
    fun showLoading(isLoading: Boolean)
    fun showFetchCompleted(images: List<Image>, folders: List<Folder>)
    fun showError(throwable: Throwable)
    fun showEmpty()
    fun showCapturedImage(images: List<Image>)
    fun finishPickImages(images: List<Image>)
}